programs/autocrat/src/instructions/initialize_proposal.rs

```diff
@@ -25,28 +25,34 @@ pub struct InitializeProposal<'info> {
     pub proposal: Box<Account<'info, Proposal>>,
     #[account(mut)]
     pub dao: Box<Account<'info, Dao>>,
+    #[account(
+        constraint = question.oracle == proposal.key()
+    )]
+    pub question: Box<Account<'info, Question>>,
     #[account(
         constraint = quote_vault.underlying_token_mint == dao.usdc_mint,
-        constraint = quote_vault.settlement_authority == proposal.key() @ AutocratError::InvalidSettlementAuthority,
+        has_one = question,
+        // constraint = quote_vault.settlement_authority == proposal.key() @ AutocratError::InvalidSettlementAuthority,
     )]
     pub quote_vault: Account<'info, ConditionalVaultAccount>,
     #[account(
         constraint = base_vault.underlying_token_mint == dao.token_mint,
-        constraint = base_vault.settlement_authority == proposal.key() @ AutocratError::InvalidSettlementAuthority,
+        has_one = question,
+        // constraint = base_vault.settlement_authority == proposal.key() @ AutocratError::InvalidSettlementAuthority,
     )]
     pub base_vault: Account<'info, ConditionalVaultAccount>,
     #[account(
-        constraint = pass_amm.base_mint == base_vault.conditional_on_finalize_token_mint,
-        constraint = pass_amm.quote_mint == quote_vault.conditional_on_finalize_token_mint,
+        constraint = pass_amm.base_mint == base_vault.conditional_token_mints[1],
```

Comment:
Instead of using 0 and 1 everywhere i'm wondering if it would be better / possible to use an enum

Reply:
100%, this is a great idea

Reply:
Addressed by 1b68f4b68dfa06fe0a670918ee7199471733672b